src/app/(public)/repos/_components/pagination.tsx

```diff
@@ -15,24 +18,69 @@ export function Pagination({
   totalCount,
   searchParams
 }: PaginationProps) {
+  const [isLoading, setIsLoading] = useState(false);
```

Comment:
Maybe let's make use of the [`useTransition` hook](https://react.dev/reference/react/useTransition) here and wrap `router.push` calls in `startTransition()`